File: hh-back/src/main/java/gloit/hiperionida/helios/model/CategoriaGastoModel.java
package gloit.hiperionida.helios.model;

import gloit.hiperionida.helios.util.model.AbsAuditoriaModel;

import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.SuperBuilder;

import java.time.LocalDateTime;

@EqualsAndHashCode(callSuper = false)
@Entity
@Getter
@Setter
@SuperBuilder
@Table(name = "categoria_gasto")
public class CategoriaGastoModel extends AbsAuditoriaModel {
    private String categoria;
    @Column(columnDefinition = "TEXT")
    private String notas;

    public CategoriaGastoModel() {
    }

    public CategoriaGastoModel(Long id, Long creadorId, LocalDateTime creada, Long modificadorId, LocalDateTime modificada, Long eliminadorId, LocalDateTime eliminada, String categoria, String notas) {
        super(id, creadorId, creada, modificadorId, modificada, eliminadorId, eliminada);
        this.categoria = categoria;
        this.notas = notas;
    }
}

File: hh-front/src/models/categoria_gasto_model.js
import { UsuarioModel } from './usuario_model'

export class CategoriaGastoModel {
  constructor () {
    this.id = ''
    this.categoria = ''

    this.creador = new UsuarioModel()
    this.creada = ''
    this.modificador = new UsuarioModel()
    this.modificada = ''
    this.eliminador = new UsuarioModel()
    this.elimianda = ''
  }
}
